File: code/data_parser.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
  FILE *fptr_in = fopen("./data/gsod.obs.csv", "r");
  char id[100];
  int wban = 0;
  int year = 0;
  int month = 0;
  int day = 0;
  int measure = 0;
  float value = 0;
  int ref = 0;
  fscanf(fptr_in, "%*[^,],%*[^,],%*[^,],%*[^,],%*[^,],%*[^,]");
  fscanf(fptr_in, "%[^,],%d,%d/%d/%d,%d,%d,%f\n", id, &wban, &year, &month, &day, &measure, &value, &ref); 
  
  printf("%s\n", id);
  return 0;
}

File: code/data_parser.py
import csv

with open('./data/gsod.obs.csv') as csvfile:
  with open('./data/us-in.station-ids', "r") as ids:
    with open('./data/us-in.station-wban', "r") as wban:
      with open('./data/good_data.csv', "a") as indiana:
        readCSV = csv.reader(csvfile, delimiter = ",")
        i = 0
        for row in readCSV:
          print(i)
          i = i + 1
          iddata = ids.readlines()  
          words = iddata[0].split("|")
          for word in words:
            if(row[0] == word):
              print(row)
              indiana.write(str(row) + "\n")
              break
          ids.seek(0)
      	  wbandata = wban.readlines()
          words = wbandata[0].split("|")
          for word in words:
            if(row[1] == word):
              print(row)
              indiana.write(str(row) + "\n")
	      break
          wban.seek(0)
